pkg/osquery/tables/tablehelpers/parser.go

```diff
@@ -0,0 +1,126 @@
+package tablehelpers
+
+import (
+	"bufio"
+	"bytes"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+type Matcher struct {
+	Match   func(string) bool
+	KeyFunc func(string) (string, error)
+	ValFunc func(string) (string, error)
+}
+
+type OutputParser struct {
+	matchers []Matcher
+	logger   log.Logger
+}
+
+func NewParser(logger log.Logger, matchers []Matcher) *OutputParser {
+	p := &OutputParser{
+		matchers: matchers,
+		logger:   logger,
+	}
+	return p
+}
+
+// Parse looks at a chunk of input. It is assumed that the input contains
+// information to fill in a single result. Do not provide input that contains
+// data for multiple results.
+func (p *OutputParser) Parse(input *bytes.Buffer) map[string]string {
```

Comment:
Yay helpers!

How hard is it to move `pkg/osquery/tables/firmwarepasswd` to using this? If it's more than a couple minutes I'm not sure we should block. Make an issue?

Reply:
I have a git stash locally that starts to move it, but it requires a slight tweak or two. I'll file an issue for it.